src/components/Inventory/withdraw/flow/SelectPurposeForm.tsx

```diff
@@ -146,25 +125,55 @@ export default function SelectPurposeForm(props: SelectPurposeFormProps): JSX.El
 
   const validateNurseryTransfer = () => {
     if (isNurseryTransfer) {
-      if (record[0].destinationFacilityId === -1) {
+      if (localRecords[0].destinationFacilityId === -1) {
         setIndividualError('destinationFacilityId', strings.REQUIRED_FIELD);
         return false;
       }
     }
     return true;
   };
 
+  const validateSelectedNursery = () => {
+    if (!selectedNursery) {
+      setIndividualError('fromFacilityId', strings.REQUIRED_FIELD);
+      return false;
+    }
+    return true;
+  };
+
   const saveWithdrawal = async () => {
-    if (record) {
+    if (localRecords) {
       const nurseryTransferInvalid = !validateNurseryTransfer();
-      if (fieldsErrors.withdrawDate || nurseryTransferInvalid) {
+      const selectedNurseryInvalid = !validateSelectedNursery();
+      if (fieldsErrors.withdrawDate || nurseryTransferInvalid || selectedNurseryInvalid) {
         return;
       }
     }
 
+    setRecords(localRecords.filter((rcd) => rcd.facilityId.toString() === selectedNursery));
+
     onNext();
   };
 
+  const onChangeFromNursery = (facilityIdSelected: string) => {
+    setSelectedNursery(facilityIdSelected);
+    setRecords(records.filter((localR) => localR.facilityId.toString() === facilityIdSelected));
+  };
+
+  const getNurseriesOptions = () => {
+    const nurseriesOptions: DropdownItem[] = [];
+    batches.forEach((bat) => {
```

Comment:
Optional suggestion (since `find` is a sequential search, we can use assoc array). Current code is just as fine.
```
const nurseries = batches.reduce((acc, batch) => {
  if (!acc[batch.facility_id.toString()]) { // this lookup may be faster
     acc[batch.facility_id.toString()] = { label: batch.facility_id, value: batch.facility_name};
  }
  return acc;
}, {});
const options = Object.values(nurseries); // or Object.keys(nurseries).map(id => nurseries[id]);
if (options.length === 1) {
  setSelectedNursery(...)
}
return options;
```